clippy_lints/src/empty_with_brackets.rs

```diff
@@ -70,10 +75,21 @@ declare_clippy_lint! {
     "finds enum variants with empty brackets"
 }
 
-declare_lint_pass!(EmptyWithBrackets => [EMPTY_STRUCTS_WITH_BRACKETS, EMPTY_ENUM_VARIANTS_WITH_BRACKETS]);
+enum Usage {
+    Unused(Span),
+    Used,
+}
+
+#[derive(Default)]
+pub struct EmptyWithBrackets {
+    // Value holds `true` if the empty tuple variant was used as a function
+    empty_tuple_enum_variants: FxIndexMap<DefId, Usage>,
```

Comment:
This can use a `LocalDefId` as key since the variants are always from the same crate (gets rid of a bunch of `.to_def_id()`s also)